File: app/src/main/java/com/jat/medilinkapp/viewmodels/NfcDataHistoryViewModel.java
package com.jat.medilinkapp.viewmodels;

import android.app.Application;

import com.jat.medilinkapp.model.MyDataBase;
import com.jat.medilinkapp.model.NfcData;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import io.reactivex.Observable;


public class NfcDataHistoryViewModel extends AndroidViewModel {

    private MyDataBase appDatabase;

    public NfcDataHistoryViewModel(@NonNull Application application)     {
        super(application);
        appDatabase = MyDataBase.getDatabase(application);
    }

    public Observable<List<NfcData>> getList() {
        return appDatabase.nfcDataDao().getNfcDataList();
    }

    long addData(NfcData modelClass) {
        return appDatabase.nfcDataDao().addData(modelClass);
    }

}

File: app/src/main/java/com/jat/medilinkapp/viewmodels/NfcDataHistoryViewModel.kt
package com.jat.medilinkapp.viewmodels

import android.app.Application

import com.jat.medilinkapp.model.NfcDataRepository
import com.jat.medilinkapp.model.entity.NfcData
import androidx.lifecycle.AndroidViewModel
import io.reactivex.Observable


class NfcDataHistoryViewModel(application: Application) : AndroidViewModel(application) {

    private var nfcDataRepository = NfcDataRepository(application)

    val list: Observable<List<NfcData>>
        get() = nfcDataRepository.list

    fun getListBySubCreateData(subDate: String): Observable<List<NfcData>> {
        return nfcDataRepository.getListBySubCreateData(subDate)
    }

    fun getListIsSend(send: Boolean): Observable<List<NfcData>> {
        return nfcDataRepository.getListIsSend(send)
    }

    fun addData(modelClass: NfcData): Long? {
        return nfcDataRepository.addData(modelClass)
    }

    fun deleteAll() {
        nfcDataRepository.deleteAll()
    }

    fun delete(nfcData: NfcData) {
        nfcDataRepository.delete(nfcData)
    }
}
